zerver/models.py

```diff
@@ -857,10 +857,14 @@ class Meta:
     @property
     def profile_data(self) -> ProfileData:
         values = CustomProfileFieldValue.objects.filter(user_profile=self)
-        user_data = {v.field_id: v.value for v in values}
+        user_data = {v.field_id: {"value": v.value, "rendered_value": v.rendered_value} for v in values}
         data = []  # type: ProfileData
         for field in custom_profile_fields_for_realm(self.realm_id):
-            value = user_data.get(field.id, None)
+            field_values = user_data.get(field.id, None)
+            if field_values:
+                value, rendered_value = field_values.get("value"), field_values.get("rendered_value")
+            else:
+                value, rendered_value = None, None
```

Comment:
To cover this case in tests, you can add new custom field in realm and don't set any value for that field for any user. And then get profile_data for that user. That way, as user don't have any value for field, it will return `None` in both case.

Reply:
Ah, thanks a lot for this advice! All of my tests finally pass :smile:.